src/operator/nn/mkldnn/mkldnn_act.cc

```diff
@@ -107,36 +107,46 @@ class MKLDNNActForward {
                    const NDArray &data, const mkldnn::memory &mem): fwd_pd(
                        GetActFwdDescImpl(param, is_train, mem, data.dtype())) {
   }
+  void SetDataHandle(const NDArray &in_data, const NDArray &out_data);
+  void Execute();
+};  // class MKLDNNActForward
 
-  void SetNewMem(const mkldnn::memory &data, const mkldnn::memory &output) {
-    if (this->data == nullptr)
+void MKLDNNActForward::SetDataHandle(const NDArray &in_data, const NDArray &out_data) {
+    auto input_mem = in_data.GetMKLDNNData();
+    auto out_mem = const_cast<NDArray &>(out_data).CreateMKLDNNData(
+        fwd_pd.dst_primitive_desc());
+
+    if (this->data == nullptr) {
       this->data = std::shared_ptr<mkldnn::memory>(new mkldnn::memory(
-              data.get_primitive_desc(), data.get_data_handle()));
-    else
-      this->data->set_data_handle(data.get_data_handle());
+              input_mem->get_primitive_desc(), input_mem->get_data_handle()));
+    } else {
+      this->data->set_data_handle(input_mem->get_data_handle());
+    }
 
-    CHECK(fwd_pd.dst_primitive_desc() == output.get_primitive_desc());
-    if (this->out == nullptr)
+    CHECK(fwd_pd.dst_primitive_desc() == out_mem->get_primitive_desc());
```

Comment:
Why check this? out_mem is just created in L116 with the dst_primitive_desc. Do you think mkldnn will fail in doing that?